CRM/Contact/Form/Search/Custom/PriceSet.php

```diff
@@ -116,19 +116,24 @@ public function fillTable() {
     CRM_Core_DAO::executeQuery($sql);
 
     $sql = "
-SELECT c.id as contact_id,
-       p.id as participant_id,
-       l.price_field_value_id as price_field_value_id,
-       l.qty
-FROM   civicrm_contact c,
-       civicrm_participant  p,
-       civicrm_line_item    l
-WHERE  c.id = p.contact_id
-AND    p.event_id = {$this->_eventID}
-AND    p.id = l.entity_id
-AND    l.entity_table ='civicrm_participant'
-ORDER BY c.id, l.price_field_value_id;
-";
+      SELECT c.id as contact_id,
+        p.id as participant_id,
+        l.price_field_value_id AS price_field_value_id,
+        l.qty
+      FROM civicrm_contact c,
+        INNER JOIN civicrm_participant p
+          ON p.contact_id = c.id AND c.is_deleted = 0
+        INNER JOIN civicrm_line_item l
+          ON p.id = l.entity_id AND l.entity_table ='civicrm_participant'
+        INNER JOIN civicrm_price_field_value cpfv
+          ON cpfv.id = l.price_field_value_id AND cpfv.is_active = 1
+        INNER JOIN civicrm_price_field cpf
+          ON cpf.id = l.price_field_id AND cpf.is_active = 1
+        INNER JOIN civicrm_price_set cps
+          ON cps.id = cpf.price_set_id AND cps.is_active = 1
+      WHERE  p.event_id = {$this->_eventID}
```

Comment:
@	pradpnayak could `$this->_eventID` be set by URL at all?

Reply:
It should be parameterised anyway I guess - it's on the form layer & the answer might be no now...

Reply:
sorry for delay in testing, finally got a chance to take a look, I received a syntax error message:

```
[nativecode=1064 ** You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'INNER JOIN civicrm_participant p
          ON p.contact_id = c.id AND c.is_delet' at line 6]
```

My SQL isn't very strong, but I believe you just need to drop the comma before the first inner join.

```
      FROM civicrm_contact c**,**
        INNER JOIN civicrm_participant p
```